### PR TITLE
MAT-5004 non owner of measure updating measure issue

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-5004](https://jira.cms.gov/browse/MAT-5004)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java [hunk 1]
```diff
@@ -121,14 +120,23 @@ public ResponseEntity<String> updateMeasure(
...
             persistedMeasure.get().getCreatedBy());
-        measureService.checkDeletionCredentials(username, persistedMeasure.get().getCreatedBy());
+        // either owner or shared-with role
+        measureService.verifyAuthorization(username, persistedMeasure.get());
+
+        // no user can update a soft-deleted measure
+        if (!persistedMeasure.get().isActive()) {
+          throw new UnauthorizedException(
+              "Measure", persistedMeasure.get().getId(), principal.getName());
+        }
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/resources/MeasureController.java:135]: Should this be part of `measureService.verifyAuthorization` method?

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java [hunk 2]
```diff
@@ -121,14 +120,23 @@ public ResponseEntity<String> updateMeasure(
...
+        // shared user should be able to edit Measure but won’t have delete access
+        if (!measure.isActive()) {
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/resources/MeasureController.java:137]: I think checking on `persistedMeasure` would be better

**Review by adongare — APPROVED** [review]

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/resources/MeasureController.java:135]: @adongare verifyAuthorization only verify is the user is the owner or has the shared-with role, but for a soft-deleted measure, no user (owner or shared-with or any other user) can modify the measure, so I think they are separate things.

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java (hunk 2 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/resources/MeasureController.java:137]: @adongare Line 137 is when a user tries to soft delete a measure. In this case, only the measure owner can do the operation, even the shared-with role user cannot soft delete a measure that the user does not own.

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java (hunk 1 above)
Comment (adongare) [reply src/main/java/cms/gov/madie/measure/resources/MeasureController.java:135]: thanks for the clarification, Cecilia!
I'm thinking users should not have access to inactive measures even though they are owned by them(they don't exist for them). Unless we have a plan to allow access to inactive measures in the future, I think this conditional check and `verifyAuthorization`  is the same thing. Correct me if I misunderstood this.

Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/resources/MeasureController.java:135]: @adongare Thanks a lot for the suggestion Ashok! I think it makes sense. Yesterday when we talked about this, Joe mentioned: "At least until we have to concept of an Admin user, or if we elect to try out an “undo delete” feature.", so I think it's possible in the future? What do you think?

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java (hunk 2 above)
Comment (adongare) [reply src/main/java/cms/gov/madie/measure/resources/MeasureController.java:137]: yeah, you're right. ignore this. thanks!

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java (hunk 1 above)
Comment (adongare) [reply src/main/java/cms/gov/madie/measure/resources/MeasureController.java:135]: I'm not sure either. I'm fine with leaving it as it is.

**Pull request merged**